packages/renderer/src/stores/build-images.ts

```diff
@@ -16,13 +16,51 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 
+import type { Terminal } from '@xterm/xterm';
 import type { Writable } from 'svelte/store';
-import { writable } from 'svelte/store';
+import { get, writable } from 'svelte/store';
+
+import type { ProviderContainerConnectionInfo } from '/@api/provider-info';
+
+export interface BuildArg {
+  key: string;
+  value: string;
+}
 
 export interface BuildImageInfo {
-  buildImageKey: symbol;
+  buildImageKey?: symbol;
   buildRunning: boolean;
+  buildFinished: boolean;
+  containerImageName: string;
+  containerFilePath: string;
+  containerBuildContextDirectory: string;
+  containerBuildPlatform: string;
+  buildParentImageName?: string;
+  buildError?: string;
+  buildArgs: BuildArg[];
+  cancellableTokenId?: number;
+  selectedProvider?: ProviderContainerConnectionInfo;
+  logsTerminal?: Terminal;
+  taskId: number;
+}
+
+export function cleanupBuildImageInfo(taskId: number): void {
+  const map = get(buildImagesInfo);
+  map.delete(taskId);
+  buildImagesInfo.set(map);
+}
+
+export function createDefaultBuildImageInfo(): BuildImageInfo {
+  return {
+    taskId: get(buildImagesInfo).size + 1,
+    containerImageName: '',
+    buildRunning: false,
+    buildFinished: false,
+    containerFilePath: '',
+    containerBuildContextDirectory: '',
+    containerBuildPlatform: '',
+    buildArgs: [{ key: '', value: '' }],
+  };
 }
 
-// current build key
-export const buildImagesInfo: Writable<BuildImageInfo> = writable();
+export const buildImagesInfo: Writable<Map<number, BuildImageInfo>> = writable(new Map());
```

Comment:
remark: I would rather have the store in a dedicated files like `build-image-info-store` or something with an explicit name, otherwise with the content of the file, and the store being at the bottom of the file, it get lost

Reply:
The file was not introduced by this fix, it should be follow up issue probably.